src/main/java/UI/DialogBox.java

```diff
@@ -0,0 +1,62 @@
+package UI;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import javafx.fxml.FXML;
+import javafx.fxml.FXMLLoader;
+import javafx.geometry.Pos;
+import javafx.scene.Node;
+import javafx.scene.control.Label;
+import javafx.scene.image.Image;
+import javafx.scene.image.ImageView;
+import javafx.scene.layout.HBox;
+
+import java.io.IOException;
+import java.util.Collections;
+
+/**
+ * Represents a dialog box consisting of an ImageView to represent the speaker's face
+ * and a label containing text from the speaker.
+ */
+public class DialogBox extends HBox {
```

Comment:
Appropriately used nouns for your classes (DialogBox, Label, ImageView) and methods that represent actions, such as getUserDialog() and getDukeDialog(). The naming is clear and descriptive, which enhances code readability.